contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,127 @@
+        uint i;
+        for (i=0; i < validators.length; i++) {
+            require(newValidator != validators[i], "validator is already active");
+        }
+        if (allowedAccounts[msg.sender].activeValidator) {
+            validators[allowedAccounts[msg.sender].validatorIndex] = newValidator;
+        } else {
+            allowedAccounts[msg.sender] = accountInfo(true, true, uint8(validators.length));
+            validators.push(newValidator);
+        }
+
+    }
+
+    function deactivate() external senderIsAllowed {
+        require(validators.length > 1, "cannot deactivate last validator");
+        require(allowedAccounts[msg.sender].activeValidator, "sender does not have an active validator");
+        allowedAccounts[msg.sender].activeValidator = false;
+        validators[allowedAccounts[msg.sender].validatorIndex] = validators[validators.length-1];
+        validators.pop();
+    }
+
+    function voteToAddAccountToAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == false, "account to add is already on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to add account");
+        }
+        currentVotes[account].push(msg.sender);
+    }
+
+    function voteToRemoveAccountFromAllowList(address account) external senderIsAllowed {
+        require(account != address(0), "account to be added cannot be 0");
+        require(allowedAccounts[account].allowed == true, "account to remove is not on the allow list");
+
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            require(currentVotes[account][i] != msg.sender, "sender has already voted to remove account");
+        }
+        currentVotes[account].push(msg.sender);
+    }
+
+    function removeVoteForAccount(address account) external senderIsAllowed {
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (currentVotes[account][i] == msg.sender) {
+                currentVotes[account][i] = currentVotes[account][currentVotes[account].length-1];
+                currentVotes[account].pop();
+                break;
+            }
+        }
+    }
+
+    function countVotes(address account) external senderIsAllowed returns(uint) {
+        uint numVotes;
+        for (uint i=0; i < currentVotes[account].length; i++) {
+            if (allowedAccounts[currentVotes[account][i]].allowed) {
+                // only increment numVotes if account that voted is still allowed
+                numVotes++;
+            }
+        }
+        if (numVotes > numAllowedAccounts / 2) {
+            delete(currentVotes[account]); // Is it a good idea to delete the array? What else could I do? Array should be max 128 long
```

Comment:
`delete` is fine. If the array is too long we could also `currentVotes.length = 0`. But that would not free the storage and I would only use this pattern when gas matters--which it does not for lengths below ~1000.